### ISSUE TITLE
[clang-format] midl generate interfaces incorrectly identify interface opening brace as FunctionLBrace

### ISSUE THREAD

**Issue by mydeveloperday** [llvm/llvm-project#58463]
Given the following code generated by midl from a com.idl, this incorrectly marks the opening `{`  as FunctionLBrace.

FunctionLBrace is the driver for RemoveSemicolons (its only as good as our FunctionLBrace detection), as such the trailing semi colon at the end of the interface will be incorrectly removed.

```c++
MIDL_INTERFACE("XXXXX-XXXXXX-XXXXXXX-XXXXXX")
IMyPersist : public IPersist
{
public:
    virtual Foo(void) = 0;
};
```

Debug trace
```
AnnotatedTokens(L=0, T=5, C=0):
 M=0 C=0 T=Unknown S=1 F=0 B=0 BK=0 P=0 Name=identifier L=10 PPK=2 FakeLParens= FakeRParens=0 II=0x162f3e082b0 Text='IMyPersist'
 M=0 C=1 T=InheritanceColon S=1 F=0 B=0 BK=0 P=23 Name=colon L=12 PPK=2 FakeLParens= FakeRParens=0 II=0x0 Text=':'
 M=0 C=0 T=Unknown S=1 F=0 B=0 BK=0 P=22 Name=public L=19 PPK=2 FakeLParens= FakeRParens=0 II=0x162f3e023e0 Text='public'
 M=0 C=0 T=Unknown S=1 F=0 B=0 BK=0 P=23 Name=identifier L=28 PPK=2 FakeLParens= FakeRParens=0 II=0x162f3e082e8 Text='IPersist'
 M=0 C=0 T=FunctionLBrace S=1 F=0 B=0 BK=1 P=23 Name=l_brace L=30 PPK=2 FakeLParens= FakeRParens=0 II=0x0 Text='{'
----
AnnotatedTokens(L=1, T=5, C=0):
 M=0 C=0 T=Unknown S=1 F=0 B=0 BK=0 P=0 Name=public L=6 PPK=2 FakeLParens= FakeRParens=0 II=0x162f3e023e0 Text='public'
 M=0 C=1 T=InheritanceColon S=0 F=0 B=0 BK=0 P=23 Name=colon L=7 PPK=2 FakeLParens= FakeRParens=0 II=0x0 Text=':'
```

**Comment by owenca** [comment]
What's `MIDL_INTERFACE("XXXXX-XXXXXX-XXXXXXX-XXXXXX")` expanded to? `class`?
